### PR TITLE
BNGP-3150 & BNGP-2886: Main Credits Journey: Credits Reference Number Generation with confirmation page

### PR DESCRIPTION
**Pull request by usman-j-shaikh**

### CODE REVIEW

File: packages/webapp/src/routes/test/seed-credits-estimation-data.js
Comment (paulsimonandrews) [inline packages/webapp/src/routes/test/seed-credits-estimation-data.js:1]: use credits-purchase not credits-estimation in filename and throughout this file

File: packages/webapp/src/utils/constants.js [hunk 1]
```diff
@@ -50,6 +51,7 @@ const ORGANISATION = 'organisation'
 const MULTIPLE_PROOFS_OF_PERMISSION_REQUIRED = 'multipleProofsOfPermissionRequired'
 const ACCESSIBILITY_STATEMENT = 'accessibility-statement'
 const COOKIES = 'cookies'
+const TEST_CREDITS_ESTIMATION_DATA = 'test/seed-credits-estimation-data'
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/constants.js:54]: CREDITS_PURCHASE not ESTIMATION

File: packages/webapp/src/utils/credits-application.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import constants from './constants.js'
+
+const getCreditsAppReference = session => session.get(constants.redisKeys.CREDITS_PURCHASE_APPLICATION_REFERENCE) || ''
+
+// Credits Estimator's object schema must match the expected payload format for the Operator application
+export default (session, account) => {
+  return {
+    creditsEstimation: {
+      applicant: {
+        firstName: account.idTokenClaims.firstName,
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/credits-application.js:10]: Won't need firstName, lastName and emailAddress in the payload so can remove them. We only need the contactId

File: packages/webapp/src/views/test/seed-credits-estimation-data.html
Comment (paulsimonandrews) [inline packages/webapp/src/views/test/seed-credits-estimation-data.html:1]: should be credits-purchase not credits-estimation

File: packages/webapp/src/views/credits-purchase/credits-confirmation.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      <h1 class="govuk-panel__title govuk-!-margin-bottom-4">
+        {{ pageHeading }}
+      </h1>
+
+      <div class="govuk-panel__body govuk-!-font-size-27">
+        Your BNG credit reference<br> <strong data-testid="application-reference-value">{{ applicationReference }}</strong>
+      </div>
+    </div>
+
+    <p>We will send you a confirmation email.</p>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-confirmation.html:21]: Doesn't match latest copydeck. Should be `We have sent you a confirmation email.`

File: packages/webapp/src/utils/credits-application-validation.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import Joi from 'joi'
+
+const creditsApplicationValidation = Joi.object({
+  creditsEstimation: Joi.object({
+    applicant: Joi.object({
+      firstName: Joi.string().required(),
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/credits-application-validation.js:6]: Won't need firstName, lastName and emailAddress (see comment below)

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Done a quick review and require some changes. Once the database migrations are sorted I will test behaviour of the code

File: packages/database-version-control/src/migrations/2023.12.01T11.49.55.create-credits-app-reference-function.sql
Comment (paulsimonandrews) [inline packages/database-version-control/src/migrations/2023.12.01T11.49.55.create-credits-app-reference-function.sql:1]: Rename this migration so it occurs after the 2024.02.15 migration that adds the `CreditsPurchase` type

**Review by paulsimonandrews — APPROVED** [review]

**Comment by shaundefra** [comment]
Tested locally

**Pull request merged**